gatsby-node.js

```diff
@@ -34,26 +61,82 @@ exports.onPostBuild = async function(
         `failed to index to Algolia. You did not give "query" to this query`
       );
     }
+    if (!Array.isArray(matchFields) || !matchFields.length) {
+      return report.panic(
+        `failed to index to Algolia. Argument matchFields has to be an array of strings`
+      );
+    }
+
     const index = client.initIndex(indexName);
-    const mainIndexExists = await indexExists(index);
-    const tmpIndex = client.initIndex(`${indexName}_tmp`);
-    const indexToUse = mainIndexExists ? tmpIndex : index;
+    /* Use temp index if main index already exists */
+    let useTempIndex = false
+    const indexToUse = await (async function(_index) {
+      if (!enablePartialUpdates) {
+        if (useTempIndex = await indexExists(_index)) {
+          return client.initIndex(`${indexName}_tmp`);
+        }
+      }
+      return _index
+    })(index)
 
-    if (mainIndexExists) {
-      setStatus(activity, `query ${i}: copying existing index`);
-      await scopedCopyIndex(client, index, tmpIndex);
```

Comment:
this code is gone now, I think that means that the enablePartialUpdates: false mode no longer works. This should be tested :) when the main index already exists